src/public-modules/Bounties/sagas.js

```diff
@@ -28,13 +31,82 @@ const {
   loadBountiesFail,
   loadBountiesSuccess,
   loadMoreBountiesFail,
-  loadMoreBountiesSuccess
+  loadMoreBountiesSuccess,
+  setSearch,
+  toggleStageFilter,
+  setStageFilter,
+  toggleDifficultyFilter,
+  setDifficultyFilter,
+  addCategoryFilter,
+  removeCategoryFilter,
+  resetFilter
 } = actions;
 
+export function* initializeFiltersFromQuery() {
+  const params = queryStringToObject(window.location.search);
+
+  const { search, bountyStage, difficulty, category } = params;
+
+  if (search) {
+    yield put(resetFilter('search'));
+    yield put(setSearch(search));
+  }
+
+  if (difficulty === '') {
+    yield put(resetFilter('difficulty'));
+  }
+
+  if (difficulty) {
+    yield put(resetFilter('difficulty'));
+    const difficulties = difficulty.split(',');
+    for (let i = 0; i < difficulties.length; i++) {
+      yield put(toggleDifficultyFilter(difficulties[i]));
+    }
+  }
+
+  if (bountyStage === '') {
+    yield put(resetFilter('stage'));
+  }
+
+  if (bountyStage) {
+    yield put(resetFilter('stage'));
+    const stages = bountyStage.split(',');
+    for (let i = 0; i < stages.length; i++) {
+      yield put(toggleStageFilter(stages[i]));
+    }
+  }
+
+  if (category === '') {
```

Comment:
Let me explain...
The reason we have this logic is to manage all edge cases.

Since we may have defaults... (For example, /explorer defaults the stage to active).  In this case, that means if someone has the url /explorer, then active will be set to true. The way to override this is by just having the field be empty: bountyStage=

Additionally, if there are any filter fields on bountyStage then we reset the filters and the default is removed... I think the logic here makes a little more sense if you just play with the UI.